packages/output/src/node/channel-server.ts

```diff
@@ -0,0 +1,112 @@
+/********************************************************************************
+ * Copyright (C) 2019 Arm and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { OutputChannelServiceClient, OutputChannelService } from '../common/output-protocol';
+
+export interface LogOutputChannel {
+    appendLine(line: string): void;
+}
+
+interface ChannelData {
+    group: string;
+    lines: string[];
+    visible: boolean;
+}
+
+const LINES_TO_KEEP = 5000;
+const REMOVAL_SIZE = 100;
+
+@injectable()
+export class ChannelServiceImpl implements OutputChannelService {
```

Comment:
please give ext specific names and avoid generic names, i.e. `OutputChannelServiceImpl`

Reply:
I've renamed this.  I also checked other names against the majority conventions and renamed OutputChannelServiceClient to OutputChannelClient.